src/modules/dashboard/sagas/task.ts

```diff
@@ -97,118 +88,51 @@ export function* fetchColonyTaskMetadata(colonyAddress: Address) {
 
 function* taskCreate({
   meta,
-  payload: { colonyAddress, domainId = ROOT_DOMAIN },
+  payload: { colonyAddress, colonyName, ethDomainId },
 }: Action<ActionTypes.TASK_CREATE>) {
   try {
-    const {
-      record: { colonyName },
-    } = yield select(colonySelector, colonyAddress);
-
-    const { walletAddress } = yield getLoggedInUser();
-
-    // NOTE: This is going to be part of the store address so we need to be careful
-    const draftId = generateUrlFriendlyId();
-    const { taskStore, commentsStore, event } = yield executeCommand(
-      createTask,
-      {
-        metadata: { colonyAddress, draftId, domainId },
-        args: { creatorAddress: walletAddress, draftId, domainId },
-      },
+    const apolloClient: ApolloClient<any> = yield getContext(
+      Context.APOLLO_CLIENT,
     );
-    const successAction: Action<ActionTypes.TASK_CREATE_SUCCESS> = {
-      type: ActionTypes.TASK_CREATE_SUCCESS,
-      payload: {
-        commentsStoreAddress: commentsStore.address.toString(),
-        colonyAddress,
-        draftId,
-        event,
-        taskStoreAddress: taskStore.address.toString(),
-        task: {
+
+    const {
+      data: { createTask },
+    } = yield apolloClient.mutate<
+      CreateTaskMutation,
+      CreateTaskMutationVariables
+    >({
+      mutation: CreateTaskDocument,
+      variables: {
+        input: {
           colonyAddress,
-          creatorAddress: walletAddress,
-          draftId,
-          domainId,
+          ethDomainId,
         },
       },
-      meta: { key: draftId, ...meta },
+      refetchQueries: [
+        { query: ColonyTasksDocument, variables: { address: colonyAddress } },
+      ],
```

Comment:
While I'm not sure how the typings will work with something like this, it would probably be a bit more performant to update the cache here instead of relying on `refetchQueries`. It'd look something like:

```suggestion
      update: (store, { data: mutationResult }) => {
        if (mutationResult) {
          const { createTask } = mutationResult;
          const data = store.readQuery<
            ColonyTasksQuery,
            ColonyTasksQueryVariables
          >({
            query: ColonyTasksDocument,
            variables: { address: colonyAddress },
          });
          if (data && data.colony) {
            data.colony.tasks = [...data.colony.tasks, createTask];
            store.writeQuery({ query: ColonyTasksDocument, data });
          }
        }
      },
      ],
```

Reply:
cc: @chmanie 

Any opinion on this?